prompts/sources.go

```diff
@@ -0,0 +1,270 @@
+package prompts
+
+import (
+	"fmt"
+	"net/url"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/huh"
+	"github.com/pkg/errors"
+	"github.com/speakeasy-api/sdk-gen-config/workflow"
+	"github.com/speakeasy-api/speakeasy/charm"
+)
+
+func getBaseSourcePrompts(currentWorkflow *workflow.Workflow, sourceName, fileLocation, authHeader, authSecret *string) []*huh.Group {
+	groups := []*huh.Group{
+		huh.NewGroup(
+			huh.NewInput().
+				Title("A name for this source:").
+				Prompt(" ").
+				Inline(true).
+				Validate(func(s string) error {
+					if _, ok := currentWorkflow.Sources[s]; ok {
+						return fmt.Errorf("a source with the name %s already exists", s)
+					}
+					return nil
+				}).
+				Value(sourceName),
+			huh.NewInput().
+				Title("What is the location of your OpenAPI document:").
```

Comment:
Let's be consistent with either labelling fields (ie `A name for this source:`) _or_ asking questions (ie `What is the location of your OpenAPI document?`). Also, when asking questions, we should use ? instead of :